### PR TITLE
Improve check_kernel_config/CONFIG_FORTIFY_SOURCE

### PR DESCRIPTION
**Pull request by Vogtinator**
- Remove useless use of cat
- The option is no longer SLE specific
- Simplify code

- Related ticket: https://progress.opensuse.org/issues/102726
- Verification run: https://openqa.opensuse.org/tests/2062545

CC @rfan1

### CODE REVIEW

**Review by DimStar77 — APPROVED** [review]

**Comment by DimStar77** [comment]
NOTE: VR failed though

# Test died: Error: please check syslog for FORTIFY failure at /var/lib/openqa/pool/4/os-autoinst-distri-opensuse/tests/security/check_kernel_config/CONFIG_FORTIFY_SOURCE.pm line 28.

File: tests/security/check_kernel_config/CONFIG_FORTIFY_SOURCE.pm [hunk 1]
```diff
@@ -11,27 +11,20 @@ use strict;
...
 
     # Check the syslog and 'dmesg' output to make sure no error or warning messages
-    my $results = script_run("dmesg | grep -i FORTIFY");
-    if (!$results) {
+    if (script_run("dmesg | grep -i FORTIFY") == 0) {
         die("Error: please check dmesg log for FORTIFY failure");
     }
-    my $results_1 = script_run("cat /var/log/messages | grep -i FORTIFY");
-    if (!$results_1) {
+    if (script_run("grep -i FORTIFY /var/log/messages") == 0) {
```
Comment (DimStar77) [inline tests/security/check_kernel_config/CONFIG_FORTIFY_SOURCE.pm:27]: this fails on TW, as we do not log outside of systemd

Comment (Vogtinator) [reply tests/security/check_kernel_config/CONFIG_FORTIFY_SOURCE.pm:27]: Already fixed, VR was outdated

**Comment by Vogtinator** [comment]
I pasted an old VR, https://openqa.opensuse.org/tests/2062550 is newer.

Meanwhile I found yet another bug though, so https://openqa.opensuse.org/tests/2062569 it is

**Pull request merged**